app/code/local/Magmodules/Webwinkelconnect/Helper/Data.php

```diff
@@ -143,6 +143,81 @@ public function syncProductReview(array $product_review): array {
         }
     }
 
+
+    public function hasConsent(Mage_Sales_Model_Order $order): bool {
+        $permission_url = 'https://dashboard.webwinkelkeur.nl/api/2.0/order_permissions.json?' .
+            http_build_query([
+                'orderNumber' => $order->getIncrementId(),
+                'id' => $this->getShopId(),
+                'code' => $this->getApiKey(),
+            ]);
+
+        $curl = new Varien_Http_Adapter_Curl();
+        $curl->setConfig([
+            'timeout'   => 10,
+        ]);
+        $curl->addOptions([
+            CURLOPT_RETURNTRANSFER => true,
+            CURLOPT_FOLLOWLOCATION => true,
+            CURLOPT_FAILONERROR => true,
+        ]);
+        $curl->write(Zend_Http_Client::GET, $permission_url, '2', false);
+        $response = $curl->read();
+        $log = Mage::getModel('webwinkelconnect/log');
+        $start_time = microtime(true);
+        $message = '';
+        $response_html = '';
+        $decoded_body = '';
+        if ($response) {
```

Comment:
The proper way to check the result from cURL is to do this:

~~~php
$response = curl_exec($curl);
if ($response === false) {
  // It failed...
}
~~~

That's because the empty string is also falsy, but doesn't actually indicate an error condition.